.github/workflows/testPublish.yml

```diff
@@ -0,0 +1,46 @@
+name: Test and publish
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the master branch
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [18.x]
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+      - run: npm ci
+      - run: npm run build
+
+  release:
+    # Only release on push to master
+    if: github.event_name == 'push' && github.ref == 'refs/heads/master'
+    runs-on: ubuntu-latest
+    needs: test
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - run: npm ci
+      - run: npm run build
+      - uses: cycjimmy/semantic-release-action@v3
```

Comment:
Should we introduce a [`.releaserc` or some sort of configuration](https://github.com/semantic-release/semantic-release/blob/master/docs/usage/configuration.md#configuration) to specify the `main` branch or any plugins we might be interested in using?

Reply:
@Izook what is the value of adding the .releaserc here? This part of the script only runs when there is a merge of a branch into the main branch.

Reply:
I noticed that this action utilizes `semantic-release` which can be configured via a `.releaserc` or a `"release"` key in the `package.json`. I'm not sure if we do but we might need it to specify that we want to release off of the `main` branch as opposed to `master`. I also just like to inclde configuration files whenever they can be read to sort of explictly say we are using the default behavior but I'm totally ok if we don't go this direction.

Reply:
Good comments @Izook. Happy to continue exploring this idea but let's do that outside of this PR since other work is blocked by this.